src/lib/storage/segment_accessor.hpp

```diff
@@ -52,37 +54,56 @@ class SegmentAccessor : public BaseSegmentAccessor<T> {
 
 /**
  * For ReferenceSegments, we don't use the SegmentAccessor but either the MultipleChunkReferenceSegmentAccessor or the.
- * SingleChunkReferenceSegmentAccessor. The first one is generally applicable. For each offset that is accessed, a new
- * accessor has to be created. This is because we cannot be sure that two consecutive offsets reference the same chunk.
- * In the SingleChunkReferenceSegmentAccessor, we know that the same chunk is referenced, so we create the accessor
- * only once.
+ * SingleChunkReferenceSegmentAccessor. The first one is generally applicable. However, we will have more overhead,
+ * because we cannot be sure that two consecutive offsets reference the same chunk. In the
+ * SingleChunkReferenceSegmentAccessor, we know that the same chunk is referenced, so we create the accessor only once.
  */
 template <typename T>
-class MultipleChunkReferenceSegmentAccessor : public BaseSegmentAccessor<T> {
+class MultipleChunkReferenceSegmentAccessor : public AbstractSegmentAccessor<T> {
  public:
-  explicit MultipleChunkReferenceSegmentAccessor(const ReferenceSegment& segment) : _segment{segment} {}
+  /**
+   * AVG_CHUNKS is a constant to estimate how many different chunks are to be expected within a ReferenceSegment.
+   * It is used if the constructor isn't called with the maximum chunk_id within the ReferenceSegment.
+   * Setting AVG_CHUNKS too small will lead to having to resize _accessors more than once.
+   * If AVG_CHUNKS is set too big, _accessors will reserve more space than necessary.
+   */
+  static constexpr ChunkID AVG_CHUNKS{10u};
```

Comment:
Does this `10` have any justification? Random magic numbers we should probably better avoid.